integrators/explicit_embedded_runge_kutta_nyström_integrator_body.hpp

```diff
@@ -0,0 +1,209 @@
+  using Displacement = Difference<Position>;
+  using Velocity = Variation<Position>;
+  using Acceleration = Variation<Velocity>;
+
+  // Argument checks.
+  int const dimension = initial_value.positions.size();
+  CHECK_EQ(initial_value.momenta.size(), dimension);
+  CHECK_NE(first_time_step, Time());
+  Sign const integration_direction = Sign(first_time_step);
+  if (integration_direction.Positive()) {
+    // Integrating forward.
+    CHECK_LT(initial_value.time.value, t_final);
+  } else {
+    // Integrating backward.
+    CHECK_GT(initial_value.time.value, t_final);
+  }
+  CHECK_GT(safety_factor, 0);
+  CHECK_LT(safety_factor, 1);
+
+  // Time step.
+  Time h = first_time_step;
+  // Current time.
+  DoublePrecision<Time> t = initial_value.time;
+
+  // Position increment (high-order).
+  std::vector<Displacement> ∆q_hat(dimension);
+  // Velocity increment (high-order).
+  std::vector<Velocity> ∆v_hat(dimension);
+  // Current position.
+  std::vector<DoublePrecision<Position>> q_hat = initial_value.positions;
+  // Current velocity.
+  std::vector<DoublePrecision<Velocity>> v_hat = initial_value.momenta;
+
+  // Difference between the low- and high-order approximations of the position.
+  std::vector<Displacement> q_error_estimate(dimension);
+  // Difference between the low- and high-order approximations of the velocity.
+  std::vector<Velocity> v_error_estimate(dimension);
+
+  // Current Runge-Kutta-Nyström stage.
+  std::vector<Position> q_stage(dimension);
+  // Accelerations at each stage.
+  // TODO(egg): this is a rectangular container, use something more appropriate.
+  std::vector<std::vector<Acceleration>> g(stages_);
+  for (auto& g_stage : g) {
+    g_stage.resize(dimension);
+  }
+
+  bool at_end = false;
+  double tolerance_to_error_ratio;
+
+  // No step size control on the first step.
+  goto runge_kutta_nyström_step;
+
+  while (!at_end) {
+    // Compute the next step with decreasing step sizes until the error is
+    // tolerable.
+    do {
+      // Adapt step size.
+      // TODO(egg): find out whether there's a smarter way to compute that root,
+      // especially if we make the order compile-time.
+      h *= safety_factor * std::pow(tolerance_to_error_ratio,
+                                    1.0 / (lower_order_ + 1));
+
+    runge_kutta_nyström_step:
+      // Termination condition.
+      Time const time_to_end = (t_final - t.value) - t.error;
+      at_end = integration_direction * h >= integration_direction * time_to_end;
+      if (at_end) {
+        // The chosen step size will overshoot.  Clip it to just reach the end,
+        // and terminate if the step is accepted.
+        h = time_to_end;
+      }
+
+      // Runge-Kutta-Nyström iteration; fills |g|.
+      for (int i = 0; i < stages_; ++i) {
+        Time const t_stage = t.value + c_[i] * h;
+        for (int k = 0; k < dimension; ++k) {
+          Acceleration ∑j_a_ij_g_jk = Acceleration();
```

Comment:
In all these declaration, you should use value initialization:

` Acceleration ∑j_a_ij_g_jk{};`